tools/ingest_historical_csv.py

```diff
@@ -21,17 +21,25 @@ def _report_failures():
         log.error("'{}' failed with status '{}'".format(k, v))
 
 
-async def _insert_df(df, session: ClientSession, url, csv, sem):  # pylint: disable=invalid-name
+async def _insert_df(df, url, csv, sem):  # pylint: disable=invalid-name
     objs = df.to_dict(orient='records')
-    for obj in objs:
-        async with sem, session.post(url, json=obj) as response:
-            log.debug('Got response {} for {}'.format(response.status, obj))
-            if response.status != 200:
-                log.error('Error response {} for {}'.format(response.status, obj))
-                _failing_list.update(dict([(csv, response.status)]))
+    tasks = []
+    async with ClientSession() as session:
```

Comment:
It would be better if single session is used for all insertion. I mean session should be created in main and argumented like previous approach.
Basically it improves the network throughput by reusing the connection cache.

Reply:
Done